### PR TITLE
Ubuntu 16.04 (and other OS) Support

### PR DESCRIPTION
**Pull request by BTPankow**
This pr adds support for newer distros which will allow sdl_core to be compiled using recent gcc versions with no warnings. sdl_core can still be compiled on the standard development environment with these changes.

See the commit messages for more information on the changes made.

### CODE REVIEW

**Comment by pestOO** [comment]
@BTPankow 
We are not using C++11 feature as ` std::unique_ptr` due to support [C++98](https://smartdevicelink.com/en/guides/core/software-architecture-document/development-view/#481-implementation-technologies).
See also list of officially support platforms: https://smartdevicelink.com/en/guides/core/software-architecture-document/development-view/#483-development-environment-and-standards

@LuxoftAKutsan, @AGaliuzov, please support with a build for Ubuntu 16.04

**Comment by pestOO** [comment]
@LuxoftAKutsan, @abyz

**Comment by JackLivio** [comment]
@pestOO When trying to compile with the flag -std=c++98 the compiler throws this error

```
/usr/include/c++/4.8/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library
support for the ISO C++ 2011 standard. This support is currently experimental, and must be enabled 
with the -std=c++11 or -std=gnu++11 compiler options.

```

along with many other errors found in profile.cc

**Comment by jacobkeeler** [comment]
@pestOO the `cstdint` header was added is C++11, and it is included/used in several places throughout sdl_core (such as [protocol_handler_settings.h](https://github.com/smartdevicelink/sdl_core/blob/master/src/components/include/protocol_handler/protocol_handler_settings.h)), meaning that the currently released version needs to be built using the C++11 standard and uses features from this standard.

**Comment by LuxoftAKutsan** [comment]
@JackLivio @jacobkeeler That right, currently we have some issues, so we can't compile with c++98. But we are removing this issues continuously and avoid adding new ones.

File: src/components/security_manager/src/crypto_manager_impl.cc [hunk 1]
```diff
@@ -129,9 +129,15 @@ bool CryptoManagerImpl::Init() {
   const SSL_METHOD* method;
 #endif
   switch (get_settings().security_manager_protocol_name()) {
+#ifdef OPENSSL_NO_SSL3
```
Comment (LuxoftAKutsan) [inline src/components/security_manager/src/crypto_manager_impl.cc]: @BTPankow please avoid duplication  ```case SSLv3:``` and put ifdef inside case

Comment (jacobkeeler) [inline src/components/security_manager/src/crypto_manager_impl.cc]: @LuxoftAKutsan The commit has been amended with this change.

I will be taking over all of BTPankow's PR's in Core from now on, so you can ping me with any more feedback you have

**Comment by jacobkeeler** [comment]
So at this point, is the reversal of the change from `auto_ptr` to `unique_ptr` all that needs to be changed? Are there any other issues with the deprecation fixes that were made in this PR?

**Comment by LuxoftAKutsan** [comment]
@jacobkeeler I don't see any restricted issues in this pull request except uniq_ptr.

**Comment by jacobkeeler** [comment]
Closing in favor of #924

**Pull request closed without merging**